include/sleip/dynamic_array.hpp

```diff
@@ -14,6 +15,20 @@
 
 namespace sleip
 {
+namespace detail
+{
+template <class A, class T, class I>
+inline void
+alloc_move_construct_n(A& a, T* p, std::size_t n, I b)
+{
+  ::boost::detail::alloc_destroyer<A, T> hold(a, p);
+  for (std::size_t& i = hold.size(); i < n; void(++i), void(++b)) {
+    std::allocator_traits<A>::construct(a, p + i, std::move_if_noexcept(*b));
+  }
+  hold.size() = 0;
+}
```

Comment:
> Shouldn't this destroy constructed elements in the case an exception is thrown (because a copy was attempted)?

It's supposed to!

Truth be told, I'm surprised I didn't get any flack for relying on a detail function :P

Saving you the trouble of looking up the impl, the class has its size updated dynamically by the for-loop. In the destructor, it runs that many `::destroy` calls. You set it to 0 at the end of the construct loop. This was so that the object destruction happens implicitly when the user catches the exception.

> Also, how does this work in the case of movable-only potentially-throwing types?

Does a straight copy. If you're okay with it, I don't mind dropping the strong exception guarantee here and leaving the source container partially moved from.